### DISCUSSION TITLE
Problems with Standalone Command application and autoload

### DISCUSSION THREAD

**Discussion by EdwinLove** [symfony/symfony#43253] [Q&A] [answered]
I am making a small CLI app using the Symfony Command component. I have built up my domain with TDD using PhpUnit and all is well and the autoloading is working fine.
I then set up my application as suggested in this article and my application.php looks like so

```
<?php

require __DIR__ . '/vendor/autoload.php';

use App\Infrastructure\Command\FooCommand;
use Symfony\Component\Console\Application;

$application = new Application();

$application->add(new FooCommand());

$application->run();
```

I then run `php application.php list` and get the following error

```
PHP Fatal error:  Uncaught Error: Class 'App\Infrastructure\Command\FooCommand' not found in /Users/edwin.love/Work/vanilla/foo/application.php:10
Stack trace:
#0 {main}
  thrown in /Users/edwin.love/Work/vanilla/foo/application.php on line 10
```

any ideas?

**Comment by michaoj** [comment]
Can you check in your composer.json that you do have the autoloading section

```json
"autoload": {
        "psr-4": {
            "App\\": "src/"
}
```

**Comment by EdwinLove** [comment]
composer.json is a follows 

``` 
  "require": {
    "phpunit/phpunit": "^9.5",
    "symfony/console": "^5.3"
  },
  "autoload": {
    "psr-4": {
      "App\\": "src/"
    }
  },
  "autoload-dev": {
    "psr-4": {
      "App\\Tests\\": "tests/"
    }
  }```

**Comment by michaoj** [comment]
Are you sure the autoload file is indeed loaded ? Did you check your command namespace?

**Reply by EdwinLove** [reply]
I have been checking and checking wondering if there was a small typo (l fo I for example) but haven't found anything yet. I've now started writing a Command test for the command and that is failing to find the Command as well so it must be something like that. Will have a bit more of a dig

**Reply by EdwinLove** [reply]
Actually Command test is now fine, I had been wiring it up wrong

**Reply by EdwinLove** [reply]
I just replaced the line adding the Command to the application and added in another line instantiating something else from within my codebase and that worked fine so must be something about the Command that is wrong.

**Comment by EdwinLove** [comment]
path to directory the file is in is `/src/Infrastructure/Command`
namespace of class is `App\Infrastructure\Command`

**Comment by EdwinLove** [comment, marked as answer]
OK, found the problem. The FooCommand had missing dependencies that are injected through the constructor.

Can't say teh error message pointed toward that at all, oh well. All good now. Thanks to those who took the time to look and try to help.
